community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -757,9 +765,12 @@ private void close( Header.Writer headerWriter ) throws IOException
 
         try
         {
-            // Perform a checkpoint before closing. If no changes has happened since last checkpoint,
-            // no new checkpoint will be created.
-            checkpoint( IOLimiter.unlimited(), headerWriter );
+            if ( health.isHealthy() )
```

Comment:
If `didCheckpoint` would instead be called e.g. `cleanShutdown` it would apply better to other label index implementations too.